pyomo/contrib/parmest/tests/test_ScenarioCreator.py

```diff
@@ -0,0 +1,117 @@
+# the matpolotlib stuff is to avoid $DISPLAY errors on Travis (DLW Oct 2018)
+try:
+    import matplotlib
+    matplotlib.use('Agg')
+except:
+    pass
+try:
+    import numpy as np
+    import pandas as pd
+    imports_not_present = False
+except:
+    imports_not_present = True
```

Comment:
John recently over-hauled/standardized checking for optional dependencies to speed up import times for Pyomo. Take a look at `parmest.py`:

```
from pyomo.common.dependencies import (
    numpy as np, numpy_available,
    pandas as pd, pandas_available,
    scipy, scipy_available,
)
parmest_available = numpy_available & pandas_available & scipy_available
```

Reply:
OK. thanks for these pointers. I think I have done all the things you and John pointed out. A couple of tests fail, but as nearly as I can tell it is not due to anything in the pull request.